### ISSUE TITLE
Media Library has Keyboard Navigation Bugs, Lacks Proper Announcement

### ISSUE THREAD

**Issue by avaaec** [pressbooks/pressbooks#1783]
### Steps to Reproduce

1. Navigate to the Add Chapter page for a book
2. Navigate to the "Add Media" button and activate it
3. Activate the "Add Media" tab and then the "Media Library" tab
4. If you have fewer than two items in your media library, upload a few files
5. Use the keyboard to navigate to the first item in your library, implemented as a checkbox.
6. Press tab once and notice that your focus moves to a "Deselect button", notice that this is redundant
6. Now, while the first attachment is selected, navigate to the second attachment and select it.
6. Notice that the first attachment is automatically deselected, but non-visual users receive no indication of this.
7. Press tab once more and find that your focus moves to the "Attachment Details" heading
8. Notice that this element has it's tab-order attribute to set to 0 when it shouldn't be in the tab order at all. Notice that because of this, there is an invisible tab stop for visual keyboard users
9. Now navigate through the details of the attachment to the bottom of the page where there is a clear link
10. After this link there is a list of check boxes (visually appearing as attachment thumbnails)
11. Trying checking one of these check boxes and notice that they are not interactable
12. Notice that there is no labeling to inform non-visual users of this.

**Expected behavior:** 
The Media Library is a great feature and should be functionally accessible for everyone. The attachment thumbnail / check boxes should be designed so that the on click functionality is clear to everyone without needing to add the "Deselect button" as a tab stop. Additionally there should never be invisible tab stops like the "Attachment Details" heading. Finally the list at the bottom of the page should not be made of check boxes if the check boxes are not interactive. 
**Actual behavior:** 
There are no problems with the media library that individually make it an inaccessible experience, but there are a lot of small issues that together make it difficult and confusing to use as an assistive tech user.

**Comment by SteelWagstaff** [comment]
We inherit this functionality from core WordPress. This issues should probably be discussed upstream.

**Comment by avaaec** [comment]
This is a UW ATS priority 2 issue

**Comment by SteelWagstaff** [comment]
@greatislander I believe that this is an upstream WordPress issue. Do you know any more about this that might be helpful?

**Comment by greatislander** [comment]
Some of these issues appear to be resolved in WordPress 5.3. See:

- https://core.trac.wordpress.org/ticket/47122
- https://core.trac.wordpress.org/ticket/47138
- https://core.trac.wordpress.org/ticket/47141
- https://core.trac.wordpress.org/ticket/47146
- https://core.trac.wordpress.org/ticket/47610

**Comment by greatislander** [comment]
Also, worth checking out the WPCampus report's section on the media interface. A lot of these issues are upstream and have been reported as such. Many have been fixed.

See:

- https://wpcampus.org/audit/
- https://core.trac.wordpress.org/query?keywords=~wpcampus-report

**Comment by connermwallace** [comment]
University of Washington Accessibility Technology Services has inspected this issue and we have verified that it has been fixed. Version: 5.3.2.
